src/plugins/util/parser.ts

```diff
@@ -0,0 +1,159 @@
+	const getters = {
+		argument: (item: { argument: Expression | null }) => item.argument,
+		arguments: (item: { arguments: Expression[] }) => item.arguments,
+		assignment: (item: AssignmentExpression) => item.right,
+		body: (item: { body: any }) => item.body,
+		call: (item: { callee: any, arguments: any[] }) => [ item.callee, ...(item.arguments || []) ],
+		'case': (item: SwitchCase) => item.consequent,
+		cases: (item: { cases: SwitchCase[] }) => item.cases,
+		conditional: (item: { alternate: Expression; consequent: Expression; }) => [ item.alternate, item.consequent ],
+		elements: (item: { elements: Expression[] }) => item.elements,
+		expression: (item: { expression: Expression }) => item.expression,
+		expressions: (item: { expressions: Expression[] }) => item.expressions,
+		loop: (item: { test: Expression; body: Statement; }) => [ item.test, item.body ],
+		object: (item: { object: Expression | null }) => item.object,
+		properties: (item: { properties: Property[] }) => item.properties,
+		value: (item: { value: Expression }) => item.value
+	};
+
+	const nextItemMap: { [key: string]: (item: any) => any; } = {
+		ArrayExpression: getters.elements,
+		AssignmentExpression: getters.assignment,
+		CallExpression: getters.call,
+		ConditionalExpression: getters.conditional,
+		DoWhileStatement: getters.loop,
+		ExpressionStatement: getters.expression,
+		IfStatement: getters.conditional,
+		MemberExpression: getters.object,
+		NewExpression: getters.arguments,
+		ObjectExpression: getters.properties,
+		Property: getters.value,
+		ReturnStatement: getters.argument,
+		SequenceExpression: getters.expressions,
+		SwitchCase: getters[ 'case' ],
+		SwitchStatement: getters.cases,
+		VariableDeclaration: (item: VariableDeclaration) => item.declarations,
+		WhileStatement: getters.loop
+	};
+
+	[
+		'ArrowExpression',
+		'ArrowFunctionExpression',
+		'BlockStatement',
+		'ForInStatement',
+		'ForOfStatement',
+		'ForStatement',
+		'FunctionDeclaration',
+		'FunctionExpression',
+		'LetStatement',
+		'Program',
+		'TryStatement'
+
+	].forEach((type: string) => {
+		nextItemMap[ type ] = getters.body;
+	});
+
+	return function (item: any): any {
+		const getter = nextItemMap[ item.type ];
+		return typeof getter === 'function' ? getter(item) : null;
+	};
+})();
+
+/**
+ * @private
+ * Determine whether the provided node is an array expression.
+ *
+ * @param item
+ * The node to test.
+ *
+ * @return
+ * `true` if the item is an array expression; false otherwise.
+ */
+export function isArrayExpression(item: any): item is ArrayExpression {
+	return item && item.type === 'ArrayExpression';
+}
+
+/**
+ * @private
+ * Determine whether the provided node is a function declaration.
+ *
+ * @param item
+ * The item to test.
+ *
+ * @return
+ * `true` if the node represents either function declaration or a function expression.
+ */
+export function isFunctionDefinition(item: any): item is BaseFunction {
+	const { type } = item;
```

Comment:
Item shouldn't be undefined unless something is wrong with the AST tree, but I added a check regardless.